### PR TITLE
Fixing silent assertions

### PR DESCRIPTION
**Pull request by taniabogatsch**
I picked this up from @Mytherin who did the initial changes. 

The PR adds an option to make DuckDB crash when an assertion is thrown. Previously, some assertions got 'swallowed' by the system, because they were converted into exceptions which then got swallowed by a `try {} catch(...) {}` block. There now is a flag that crashes the system on assert instead, and there is also a new test for that in the CI.

Most fixes are related to passing on the correct vector type (`CONSTANT_VECTOR` for constant inputs) after executing a scalar function. This also fixes some other minor issues.

### CODE REVIEW

File: src/function/scalar/list/list_sort.cpp [hunk 1]
```diff
@@ -106,7 +106,12 @@ static void ListSortFunction(DataChunk &args, ExpressionState &state, Vector &re
 	result.SetVectorType(VectorType::FLAT_VECTOR);
 	auto &result_validity = FlatVector::Validity(result);
 
-	args.Flatten();
+	for (auto &v : args.data) {
+		if (v.GetVectorType() == VectorType::DICTIONARY_VECTOR) {
```
Comment (samansmink) [inline src/function/scalar/list/list_sort.cpp:110]: Maybe this should check for `!=FLAT_VECTOR && !=CONSTANT_VECTOR` instead as you did in `src/function/scalar/list/array_slice.cpp`? There could be other vector types in the future that need to be flattened here? (compressed vectors for example)

**Review by samansmink — COMMENTED** [review]
Added small comment, the rest LGTM!

File: src/function/scalar/list/list_sort.cpp (hunk 1 above)
Comment (taniabogatsch) [reply src/function/scalar/list/list_sort.cpp:110]: Good point!

**Comment by hannes** [comment]
There is still the failing test, can you have a look please?

**Pull request merged**